### PR TITLE
fix validate violations is swallowed by ValidationException for issue#10415

### PR DESCRIPTION
**Pull request by sparkzxl**
## What is the purpose of the change

fix validate violations is swallowed by ValidationException

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by sparkzxl** [comment]
fix issue#10415

File: dubbo-filter/dubbo-filter-validation/src/main/java/org/apache/dubbo/validation/support/jvalidation/JValidator.java [hunk 1]
```diff
@@ -275,30 +275,30 @@ public void validate(String methodName, Class<?>[] parameterTypes, Object[] argu
...
-            }
-
-            if (!violations.isEmpty()) {
-                logger.info("Failed to validate service: " + clazz.getName() + ", method: " + methodName + ", cause: " + violations);
-                throw new ConstraintViolationException("Failed to validate service: " + clazz.getName() + ", method: " + methodName + ", cause: " + violations, violations);
+                validate(violations, arg, classGroups);
             }
         } catch (ValidationException e) {
             // only use exception's message to avoid potential serialization issue
             throw new ValidationException(e.getMessage());
```
Comment (AlbumenJ) [inline dubbo-filter/dubbo-filter-validation/src/main/java/org/apache/dubbo/validation/support/jvalidation/JValidator.java:290]: `javax.validation.ConstraintViolationException` might have potential serialization issue. Should be converted.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by liufeiyu1002** [comment]
`ConstraintViolationImpl `不同版本字段不一样， 新版本新增了字段，如果单纯这么包装的话 意义不大，只是拿到了你认为的有效信息，  最起码新的结果要实现 `ConstraintViolation` 接口

**Comment by chickenlj** [comment]
Check conclustion in https://github.com/apache/dubbo/issues/10415

**Pull request closed without merging**